pkg/route/api/validation/validation.go

```diff
@@ -125,18 +125,18 @@ func ExtendedValidateRoute(route *routeapi.Route) field.ErrorList {
 	//       break, so disable the hostname validation for now.
 	// hostname := route.Spec.Host
 	hostname := ""
-	var certPool *x509.CertPool
+	var verifyOptions *x509.VerifyOptions
 
 	if len(tlsConfig.CACertificate) > 0 {
-		certPool = x509.NewCertPool()
+		certPool := x509.NewCertPool()
 		if ok := certPool.AppendCertsFromPEM([]byte(tlsConfig.CACertificate)); !ok {
 			result = append(result, field.Invalid(tlsFieldPath.Child("caCertificate"), tlsConfig.CACertificate, "failed to parse CA certificate"))
 		}
-	}
 
-	verifyOptions := &x509.VerifyOptions{
-		DNSName: hostname,
-		Roots:   certPool,
+		verifyOptions = &x509.VerifyOptions{
+			DNSName: hostname,
+			Roots:   certPool,
+		}
 	}
```

Comment:
and I would add two tests with cert->intermediate->root chains:
* include the intermediate cert in the Certificate data
* include the intermediate cert in the CACertificate data

(examples of intermediate chains at https://github.com/openshift/origin/tree/master/pkg/auth/authenticator/request/x509request/testdata)